app/assets/javascripts/angular/controller/header-controller.js

```diff
@@ -1,11 +1,21 @@
-app.controller('HeaderCtrl', ['$scope', '$http', '$routeParams', '$location', 'UserServices', function($scope, $http, $routeParams, $location, UserServices){
+app.controller('HeaderCtrl', ['$scope', '$http', '$rootScope', '$routeParams', '$location', 'UserServices', function($scope, $http, $rootScope, $routeParams, $location, UserServices){
   
   console.log("im in header ctrl");
 
   $scope.user = UserServices;
+
+
+  // watch and emit functions for UserService
+  $rootScope.$on('userDetails', function(event, args) {
+    console.log("In watch function header controller, args: ", args);
+    $scope.user = args;
```

Comment:
you could rename the parameter `args` to `user` so that we understand that what we're receiving here is an object with all the user information